json_test.go

```diff
@@ -0,0 +1,75 @@
+package matrixfederation
+
+import (
+	"testing"
+)
+
+func testSortJSON(t *testing.T, input, want string) {
+	got, err := SortJSON([]byte(input), nil)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(got) != want {
+		t.Errorf("CompactJSON(%q): want %q got %q", input, want, got)
+	}
+}
+
+func TestSortJSON(t *testing.T) {
+	testSortJSON(t, `[{"b":"two","a":1}]`, `[{"a":1,"b":"two"}]`)
```

Comment:
Cheeky tests to avoid having to check for spaces, since your sort is also removing whitespace. Mentioning that somewhere (like the `SortJSON` docs...) would be nice.